core/src/main/java/org/mskcc/cbio/oncokb/util/MainUtils.java

```diff
@@ -860,4 +861,44 @@ public static Integer compareAnnotationSearchQueryType(AnnotationSearchQueryType
             return asc ? -1 : 1;
         return (PRIORITIZED_QUERY_TYPES.indexOf(q1) - PRIORITIZED_QUERY_TYPES.indexOf(q2)) * (asc ? 1 : -1);
     }
+
+    public static int findDigitEndIndex(String str, int startIndex) {
+        int i = startIndex;
+        while (i < str.length() && Character.isDigit(str.charAt(i))) {
+            i++;
+        }
+        return i;
+    }
+
+    public static GenomicLocation parseChromosomeAndRangeFromHGVSg(String hgvsg) {
+        if (hgvsg == null) {
+            return null;
+        }
+        GenomicLocation location = new GenomicLocation();
+        int start, end;
+
+        // Step 1: Split by ":g."
+        String[] parts = hgvsg.split(":g\\.");
```

Comment:
Does parts guarantee to have two elements?

Reply:
Updated condition to check if both parts are available